C compiler warnings

as reported in https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/b020f73927a81364b1fb3d18fcb30ce26d84d6b4/variant/distributions,fedora-40-ocaml-4.14,xen-gnt-unix.4.0.2

```
#=== ERROR while compiling xen-gnt-unix.4.0.2 =================================#
# context              2.2.0~beta3~dev | linux/x86_64 | ocaml-base-compiler.4.14.2 | pinned(https://github.com/mirage/ocaml-gnt/releases/download/v4.0.2/xen-gnt-4.0.2.tbz)
# path                 ~/.opam/4.14/.opam-switch/build/xen-gnt-unix.4.0.2
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p xen-gnt-unix -j 255
# exit-code            1
# env-file             ~/.opam/log/xen-gnt-unix-7-da2d8a.env
# output-file          ~/.opam/log/xen-gnt-unix-7-da2d8a.out
### output ###
# File "lib/dune", line 12, characters 10-22:
# 12 |  (c_names gntshr_stubs gnttab_stubs)
#                ^^^^^^^^^^^^
# (cd _build/default/lib && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -g -I /home/opam/.opam/4.14/lib/ocaml -I /home/opam/.opam/4.14/lib/bytes -I /home/opam/.opam/4.14/lib/lwt -I /home/opam/.opam/4.14/lib/lwt/unix -I /home/opam/.opam/4.14/lib/ocaml/threads -I /home/opam/.opam/4.14/lib/ocplib-endian -I /home/opam/.opam/4.14/lib/ocplib-endian/bigstring -o gntshr_stubs.o -c gntshr_stubs.c)
# gntshr_stubs.c: In function 'gntshr_missing':
# gntshr_stubs.c:47:20: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
#    47 |         value *v = caml_named_value("gntshr.missing");
#       |                    ^~~~~~~~~~~~~~~~
# gntshr_stubs.c: In function 'stub_gntshr_share_pages_batched':
# gntshr_stubs.c:54:18: error: passing argument 1 of 'failwith_xc' from incompatible pointer type [-Wincompatible-pointer-types]
#    54 | #define _G(__g) ((xc_gntshr *)(__g))
#       |                 ~^~~~~~~~~~~~~~~~~~~
#       |                  |
#       |                  xc_gntshr * {aka struct xengntdev_handle *}
# gntshr_stubs.c:133:29: note: in expansion of macro '_G'
#   133 |                 failwith_xc(_G(xgh));
#       |                             ^~
# gntshr_stubs.c:60:39: note: expected 'xc_interface *' {aka 'struct xc_interface_core *'} but argument is of type 'xc_gntshr *' {aka 'struct xengntdev_handle *'}
#    60 | static void failwith_xc(xc_interface *xch)
#       |                         ~~~~~~~~~~~~~~^~~
# gntshr_stubs.c:147:20: warning: "alloc_bigarray_dims" is deprecated: use "caml_ba_alloc_dims" instead
#   147 |         ml_map = alloc_bigarray_dims(XC_GNTTAB_BIGARRAY, 1,
#       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                   
# gntshr_stubs.c: In function 'stub_gntshr_munmap_batched':
# gntshr_stubs.c:166:20: warning: "Bigarray_val" is deprecated: use "Caml_ba_array_val" instead
#   166 |         int size = Bigarray_val(ml_map)->dim[0];
#       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                      
# gntshr_stubs.c:171:20: warning: "Data_bigarray_val" is deprecated: use "Caml_ba_data_val" instead
#   171 |         int result = munmap(Data_bigarray_val(ml_map), size);
#       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                             
# gntshr_stubs.c:54:18: error: passing argument 1 of 'failwith_xc' from incompatible pointer type [-Wincompatible-pointer-types]
#    54 | #define _G(__g) ((xc_gntshr *)(__g))
#       |                 ~^~~~~~~~~~~~~~~~~~~
#       |                  |
#       |                  xc_gntshr * {aka struct xengntdev_handle *}
# gntshr_stubs.c:176:29: note: in expansion of macro '_G'
#   176 |                 failwith_xc(_G(xgh));
#       |                             ^~
# gntshr_stubs.c:60:39: note: expected 'xc_interface *' {aka 'struct xc_interface_core *'} but argument is of type 'xc_gntshr *' {aka 'struct xengntdev_handle *'}
#    60 | static void failwith_xc(xc_interface *xch)
#       |                         ~~~~~~~~~~~~~~^~~
```

--- a/lib/gntshr_stubs.c
+++ b/lib/gntshr_stubs.c
@@ -44,7 +44,7 @@ CAMLprim value stub_gntshr_allocates(void)
 
 static void gntshr_missing()
 {
-	value *v = caml_named_value("gntshr.missing");
+	const value *v = caml_named_value("gntshr.missing");
 	/* if v is NULL then it's either an error in gntshr.ml or a
 	   linking error */ 
 	assert (v != NULL);
@@ -54,27 +54,6 @@ static void gntshr_missing()
 #define _G(__g) ((xc_gntshr *)(__g))
 
 #define XC_GNTTAB_BIGARRAY (CAML_BA_UINT8 | CAML_BA_C_LAYOUT | CAML_BA_EXTERNAL)
-
-#ifdef HAVE_GNTSHR
-#define ERROR_STRLEN 1030
-static void failwith_xc(xc_interface *xch)
-{
-        static char error_str[ERROR_STRLEN];
-        if (xch) {
-                const xc_error *error = xc_get_last_error(xch);
-                if (error->code == XC_ERROR_NONE)
-                        snprintf(error_str, ERROR_STRLEN, "%d: %s", errno, strerror(errno));
-                else
-                        snprintf(error_str, ERROR_STRLEN, "%d: %s: %s",
-                                 error->code,
-                                 xc_error_code_to_desc(error->code),
-                                 error->message);
-        } else {
-                snprintf(error_str, ERROR_STRLEN, "Unable to open XC interface");
-        }
-        caml_failwith(error_str);
-}
-#endif
 
 CAMLprim value stub_gntshr_open(value unit)
 {
@@ -130,7 +109,7 @@ CAMLprim value stub_gntshr_share_pages_batched(value xgh, value domid, value cou
 
 	if(NULL == map) {
 		free(refs);
-		failwith_xc(_G(xgh));
+		caml_failwith("Failed to share pages");
 	}
 
 	// Construct the list of grant references.
@@ -144,7 +123,7 @@ CAMLprim value stub_gntshr_share_pages_batched(value xgh, value domid, value cou
 		ml_refs = ml_refs_cons;
 	}
 
-	ml_map = alloc_bigarray_dims(XC_GNTTAB_BIGARRAY, 1,
+	ml_map = caml_ba_alloc_dims(XC_GNTTAB_BIGARRAY, 1,
                               map, (c_count << XC_PAGE_SHIFT));
 
 	Store_field(result, 0, ml_refs);
@@ -163,17 +142,17 @@ CAMLprim value stub_gntshr_munmap_batched(value xgh, value share) {
 #ifdef HAVE_GNTSHR
 	ml_map = Field(share, 1);
 
-	int size = Bigarray_val(ml_map)->dim[0];
+	int size = Caml_ba_array_val(ml_map)->dim[0];
 	int pages = size >> XC_PAGE_SHIFT;
 #ifdef linux
 	/* Bug in xen-4.4 libxc xc_linux_osdep implementation, work-around
 	   by using the kernel interface directly. */
-	int result = munmap(Data_bigarray_val(ml_map), size);
+	int result = munmap(Caml_ba_data_val(ml_map), size);
 #else
-	int result = xc_gntshr_munmap(_G(xgh), Data_bigarray_val(ml_map), pages);
+	int result = xc_gntshr_munmap(_G(xgh), Caml_ba_data_val(ml_map), pages);
 #endif
 	if(result != 0)
-		failwith_xc(_G(xgh));
+		caml_failwith("Failed to munmap pages");
 #else
 	gntshr_missing();
 #endif

